collectors/AlertCollector.py

```diff
@@ -72,12 +69,16 @@ def collect(self):
         yield alert_metric
 
     def generate_alert_label_values(self, alert):
+        alert_id = alert.get('alertDefinitionId', {})
         alert_labels = dict()
-        alert_entry = self.alertdefinitions.get(alert.get('alertDefinitionId', {}), {})
+        alert_entry = self.get_alertdefinition(
+            alert_id) if alert_id not in self.alert_entry_cache else self.alert_entry_cache.get(alert_id)
```

Comment:
```suggestion
        alert_entry = self.alert_entry_cache.get(alert_id, self.get_alertdefinition(alert_id))
```

Simplifying

Reply:
`self.alert_entry_cache[alert_id]` needs to be filled here with alert_entry here to have something in there.